Modules/_json.c

```diff
@@ -628,40 +634,37 @@ py_encode_basestring(PyObject* Py_UNUSED(self), PyObject *pystr)
 }
 
 static void
-scanner_dealloc(PyObject *self)
+scanner_dealloc(PyScannerObject *self)
```

Comment:
I'm not sure why you replace `PyObject *self` with `PyScannerObject *self`. No PyScannerObject field is accessed here.